lib/matplotlib/tests/test_pyplot.py

```diff
@@ -16,7 +17,19 @@ def test_pyplot_up_to_date():
     try:
         subprocess.run([sys.executable, str(gen_script)], check=True)
         new_contents = Path(plt.__file__).read_text()
-        assert orig_contents == new_contents
+
+        if orig_contents != new_contents:
+            diff_msg = '\n'.join(
+                difflib.unified_diff(
+                    orig_contents.split('\n'), new_contents.split('\n'),
+                    fromfile='found pyplot.py',
+                    tofile='expected pyplot.py',
+                    n=0, lineterm=''))
+            raise AssertionError(
+                "pyplot.py is not up-to-date. Please run "
+                "'python tools/boilerplate.py' to update pyplot.py. "
```

Comment:
Note that this typically needs to be done *from an environment where the current matplotlib is installed* (e.g. `pip install -e`'d).

Reply:
> Should we modify boilerplate.py to import the adjacent matplotlib, even if it is not installed? IMO it makes sense to import local if we want to update the local pyplot.

But you have no guarantees that e.g. the extension modules are built, so you can fail with ImportError.  Seems easier to just ask for it to be installed...